x-pack/plugins/security_solution/public/common/lib/telemetry/events/data_quality/types.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RootSchema } from '@kbn/analytics-client';
+import type { TelemetryEventTypes } from '../../constants';
+
+export interface ReportDataQualityIndexCheckedParams {
```

Comment:
Consider something like:

```ts
export type ReportDataQualityIndexCheckedParams = ReportDataQualityCheckAllCompletedParams & {
  errorCount?: number;
  indexId: string;
  ilmPhase?: string;
  unallowedMappingFields?: string[];
  unallowedValueFields?: string[];
};
```

because it appears `ReportDataQualityIndexCheckedParams` is a superset of `ReportDataQualityCheckAllCompletedParams`